test/attr/attr_autoclosure.swift

```diff
@@ -68,40 +68,22 @@ struct AutoclosureEscapeTest {
   @autoclosure let delayed: () -> Int  // expected-error {{attribute can only be applied to types, not declarations}}
 }
 
-// @autoclosure(escaping)
+// @autoclosure(escaping) is no longer a thing; just make sure we don't crash
 // expected-error @+1 {{attribute can only be applied to types, not declarations}}
 func func10(@autoclosure(escaping _: () -> ()) { } // expected-error{{expected parameter name followed by ':'}}
 
-func func11(_: @autoclosure(escaping) @noescape () -> ()) { } // expected-error{{@escaping conflicts with @noescape}}
-  // expected-error@-1{{@autoclosure(escaping) has been removed; use @autoclosure @escaping instead}} {{28-38= @escaping}}
-
-class Super {
-  func f1(_ x: @autoclosure(escaping) () -> ()) { }
-    // expected-error@-1{{@autoclosure(escaping) has been removed; use @autoclosure @escaping instead}} {{28-38= @escaping}}
-  func f2(_ x: @autoclosure(escaping) () -> ()) { } // expected-note {{potential overridden instance method 'f2' here}}
-    // expected-error@-1{{@autoclosure(escaping) has been removed; use @autoclosure @escaping instead}} {{28-38= @escaping}}
-  func f3(x: @autoclosure () -> ()) { }
-}
-
-class Sub : Super {
-  override func f1(_ x: @autoclosure(escaping)() -> ()) { }
-    // expected-error@-1{{@autoclosure(escaping) has been removed; use @autoclosure @escaping instead}} {{37-47= @escaping }}
```

Comment:
Made it would make sense to keep parser logic to detect `@autoclosure(escpaing)` and produce old diagnostic here instead of trying to parse `escaping` as a type associated with `@autoclosure` attribute?

Reply:
Well, `@autoclosure(escaping)` was replaced with `@autoclosure @escaping` in Swift 3 which was released in 2016... and I don't think `@autoclosure(escaping)` was that common in Swift 2 code to begin with. I think it would be better for the long-term health of the code base if we drew a line in the sand and removed migration fixits for Swift 2 or older.

Reply:
Fair enough, as an alternative, maybe it would make sense to generalize this logic a bit and diagnose this problem as `@autoclosure attribute does not support arguments` to make it less confusing?